### DISCUSSION TITLE
商业源码协议为何得到 HashiCorp 等企业的垂青？ | 夜天之书

### DISCUSSION THREAD

**Discussion by giscus** [tisonkun/blog#110] [Comments]
# 商业源码协议为何得到 HashiCorp 等企业的垂青？ | 夜天之书

当地时间 8 月 10 日，知名开源软件公司 HashiCorp 发布一则公告，称其原先在 Mozilla Public License 2.0 下发布的 Terraform、Consul 和 Vault 等多款软件，将在未来的版本中改为使用商业源码协议，即 Business Source License 1.1 来发布。 此前，已有其他知名开源软件公司也改用或从一开始就使用商业源码协议许可其源码

https://www.tisonkun.org/2023/08/12/bsl/?highlight=%E5%BC%80%E6%BA%90%E5%8D%8F%E8%AE%AE

**Comment by Ray-Eldath** [comment]
想问下「作为反制，Red Hat 虽然不能修改 Linux 的软件协议，但是通过改变其发行版的打包流水线，实际上增加了云厂商在其企业级发行版上原封不动打包交付的难度。」这句有资料可以参考一下么？好奇是怎么做到的🤔

**Reply by tisonkun** [reply]
* https://www.redhat.com/en/blog/centos-linux-going-end-life-what-does-mean-me
* https://www.redhat.com/en/blog/furthering-evolution-centos-stream
* https://www.redhat.com/en/blog/red-hats-commitment-open-source-response-gitcentosorg-changes

我记得当时有人做了一个图示，大意是之前 CentOS 是慢发版有保障的，现在 CentOS Stream 是滚动发版的。商业版虽然因为 GPL 还是要共享源码，但是没有直接公开发布，得另外走个流程获取。这就改变了之前其他下游直接从 CentOS 打包的流水线。性质上没变，但是会产生一些额外的 burden
